src/mumble/AudioConfigDialog.cpp

```diff
@@ -511,21 +506,38 @@ void AudioInputDialog::updateEchoEnableState() {
 		outputInterface = s.qsAudioOutput;
 	}
 
-	if (air->canEcho(outputInterface)) {
+	qcbEcho->clear();
+	qcbEcho->setToolTip(QObject::tr("If enabled this tries to cancel out echo from the audio stream."));
+	qcbEcho->setCurrentIndex(0);
+
+	bool hasUsableEchoOption = false;
+
+	qcbEcho->insertItem(0, tr("Disabled"), "disabled");
+	qcbEcho->setItemData(0, tr("Disable echo cancellation."), Qt::ToolTipRole);
+
+	int i = 0;
+	for (EchoCancelOptionID ecoid : air->echoOptions) {
+		if (air->canEcho(ecoid, outputInterface)) {
+			++i;
+			hasUsableEchoOption = true;
+			EchoCancelOption echoOption = echoCancelOptions[static_cast<int>(ecoid)];
```

Comment:
Is making a copy here required?
From what I see we could also use a const ref instead:
```suggestion
			const EchoCancelOption &echoOption = echoCancelOptions[static_cast<int>(ecoid)];
```